### ISSUE TITLE
Mounting additionalTrustBundle in the operator.

### ISSUE THREAD

**Issue by trexx** [openshift/cluster-image-registry-operator#630]
Since 4.5.x, we've setup our cluster registry to use an internal S3 compatible endpoint for storage over HTTPS signed with our own internal CA.
We have our CAs added to additionalTrustBundle (during install) or the "user-ca-bundle" configmap in openshift-config so any cluster generated CA trust bundle has our CAs inside.
This worked with no problem.

Since upgrading to 4.6, it seems the operator does not correctly mount or trust the clusters CA trust bundle.
The CA is mounted in "/var/run/configmaps/trusted-ca/" but "/etc/pki/ca-trust/extracted/openssl/ca-bundle.trust.crt" still contains the default container image CA bundle.
This causes the errors in the operator logs and prevents our cluster upgrade from succeeding:

```
E1027 14:04:54.701531       1 controller.go:330] unable to sync: unable to sync storage configuration: RequestError: send request failed
caused by: Head "https://s3.ceph.nix.dac.local/ocps.registry": x509: certificate signed by unknown authority, requeuing
```

Opening a terminal and using curl on that endpoint also fails.

Is this intentional? I understand using alternative S3 implementations is unsupported, however this seems unrelated to S3 functionality and more to do with good ole certificate bundles and trust.

**Related past issues:**
https://github.com/openshift/cluster-image-registry-operator/issues/485
https://github.com/openshift/cluster-image-registry-operator/issues/397

**Comment by dmage** [comment]
@trexx, the CA should be in `/etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem`, it should be copied by [the operator's entrypoint](https://github.com/openshift/cluster-image-registry-operator/blob/f0425485bf830e5327a95f2af1fe8eab08115481/images/bin/entrypoint.sh#L9). Can you check this file?

**Comment by trexx** [comment]
Hi @dmage 
Running `diff /var/run/configmaps/trusted-ca/tls-ca-bundle.pem /etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem` I can see they are two different files and my internal CA certificates are not in the containers trust store.

**Comment by dmage** [comment]
That's exciting. What image reference is used by the cluster-image-registry-operator pod?

**Comment by trexx** [comment]
cluster-image-registry-operator - quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:e2b3f973bc5b9e55d2240a556c4648c921a3c8d3e12381757f1990a864208617

Correction to my original post, I mentioned 4.6, its actually 4.6.1. Sorry.

**Comment by dmage** [comment]
What are the first lines of the image registry operator's logs? Is there `Overwriting root TLS certificate authority trust store`? Are there any errors from `cp`?

**Comment by trexx** [comment]
First lines of the logs are:
```
I1027 16:23:54.997801       1 observer_polling.go:159] Starting file observer
I1027 16:23:55.053204       1 leaderelection.go:243] attempting to acquire leader lease  openshift-image-registry/openshift-master-controllers...
I1027 16:23:55.068178       1 leaderelection.go:253] successfully acquired lease openshift-image-registry/openshift-master-controllers
```
I can't see that echo, or any errors from cp.

**Comment by dmage** [comment]
Is it a new cluster? In theory the operator might be created for the configmap. If you delete the operator's pod, do you see the same logs?

**Comment by trexx** [comment]
It was an existing 4.5.16 cluster upgrading to a 4.6.1. I've deleted the pod a few times and it results in the same logs.

**Comment by trexx** [comment]
I'll prep a cluster directly to 4.6.1 to see if that makes any difference and can retry a 4.5.16 to 4.6.1 upgrade again.

**Comment by themoosman** [comment]
I'm seeing the exact same issue in a fresh 4.6.1 cluster.

**Comment by dmage** [comment]
I can confirm the problem and have created a bug in Bugzilla for that: https://bugzilla.redhat.com/show_bug.cgi?id=1892799. We'll backport it to 4.6.z. I'm closing this issue in favor of the BZ.

**Comment by trexx** [comment]
Just to add here, the fix was added in v4.6.6.

**Comment by themoosman** [comment]
Cool, I just upgraded my test cluster to 4.6.6 last evening.  I'll test it out today.
